nginx.tmpl

```diff
@@ -379,12 +382,18 @@ server {
 	{{ end }}
 
 	{{ if eq $nPaths 0 }}
-		{{ template "location" (dict "Path" "/" "Proto" $proto "Upstream" $upstream_name "Host" $host "Vhostroot" $vhost_root) }}
+		{{ template "location" (dict "Path" "/" "Proto" $proto "Upstream" $upstream_name "Host" $host "Vhostroot" $vhost_root "Dest" "") }}
 	{{ else }}
 		{{ range $path, $container := $paths }}
 			{{ $sum := sha1 $path }}
 			{{ $upstream := printf "%s-%s" $upstream_name $sum }}
-			{{ template "location" (dict "Path" $path "Proto" $proto "Upstream" $upstream "Host" $host "Vhostroot" $vhost_root) }}
+			{{ $dest := (or (first (groupByKeys $container "Env.VIRTUAL_DEST")) "") }}
```

Comment:
```suggestion
			{{ $dest := (or (first (groupByKeys $container "Env.VIRTUAL_DEST")) $path) }}
```

I think this will create the behavior you expected @buchdag

Reply:
@junderw This change is unnecessary and can cause problems. Setting $dest to $path will rewrite the matched location path with $path. This is unnecessary and breaks if $path is an regex. An empty $dest will not do any rewriting.

Reply:
I agree. Let's keep it as is.